python_modules/dagster/dagster/_components/__init__.py

```diff
@@ -232,6 +260,18 @@ def __repr__(self):
         return f"<ComponentRegistry {list(self._components.keys())}>"
 
 
+class ComponentLoadContext:
+    def __init__(self, resources: Mapping[str, object] = {}):
+        self.resources = resources
+
+
+class ComponentInitContext:
+    registry: ComponentRegistry
+
+    def __init__(self):
+        self.registry = ComponentRegistry()
+
```

Comment:
seems like we should be using `record` or `dataclass` for our own internal objects

Reply:
agree, but leaving this for now out of laziness, as these classes are certainly going to change a bunch in the near future